components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/impl/DiscoverableApplicationManagerImpl.java

```diff
@@ -34,8 +36,20 @@ public List<ApplicationBasicInfo> getDiscoverableApplicationBasicInfo(int limit,
             filter, String sortOrder, String sortBy, String tenantDomain) throws
             IdentityApplicationManagementException {
 
-        ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
-        return appDAO.getDiscoverableApplicationBasicInfo(limit, offset, filter, sortOrder, sortBy, tenantDomain);
+        try {
+            ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
+            if (ApplicationMgtUtil.isSubOrg(tenantDomain)) {
```

Comment:
Wny requirement to wrap this OrganizationManagementUtil.isOrganization(tenantDomain) with using an another util function?
Since the OrganizationManagementException is also handled here, we can directly use OrganizationManagementUtil.isOrganization(tenantDomain)